web/src/views/InvestigationDetail.vue

```diff
@@ -189,19 +190,30 @@ export default defineComponent({
     });
 
     const selectedPins: Ref<Pin[]> = computed(() => store.state.selectedPins);
-    watch(selectedPins, (pinList) => {
-      if (!featureLayer) {
-        featureLayer = createLayer('feature', { features: ['point', 'line', 'polygon'] });
-      }
-      const pinFeatureData = pinList.map((pin) => {
-        const pinLocation: Point = postGisToPoint(pin.child_location) || { x: 0, y: 0 };
+    function getPinsToDisplay() {
+      // TODO: as we move towards embedding multiple datasets into the view,
+      // we will need a more sophisticated way to determine which pins to render
+      // and determining where they should be rendered
+      const selectedPinsForActiveDataset = selectedPins.value.filter(
+        (pin) => pin.parent === activeDataset.value?.id,
+      );
```

Comment:
How would you end up selecting pins that aren't part of the active dataset? Aren't all of the pins in an investigation part of a dataset that would be active? Or is that something that might be true after our "reinterpretation" of `Investigation.datasets` as per #110?

Reply:
The pins that show up in the sidebar are related to the current investigation. `activeDataset` here is the dataset that is currently displayed by `geojs`. It is possible that pins for an investigation have different parent datasets. Since we can only show one dataset at a time (at least in the `InvestigationDetail`) right now, its possible to select pins that aren't related to the currently visible dataset.

The original change prevented such selected pins from showing up on the map, but perhaps a better change is 072af39, which disabled selection for pins whose parent dataset is not the active dataset.

Once we implement our new interpretation of `Investigation.datasets`, we'll need to update how we determine which pins are eligible to be selected, and how those pins are displayed on the map. (might be with a feature layer per tile layer, or doing math to place the pins on embedded datasets)